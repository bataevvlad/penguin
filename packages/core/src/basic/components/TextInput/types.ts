import {
  TextInputProps as DefaultTextInputProps,
  TextInput as DefaultTextInput,
} from 'react-native';

export type ITextInputProps = DefaultTextInputProps;
export type ITextInput = DefaultTextInput;
