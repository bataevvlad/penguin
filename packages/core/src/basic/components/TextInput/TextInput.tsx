import {forwardRef} from 'react';
import {TextInput as DefaultTextInput} from 'react-native';

import {getConfig} from '../../config';

import {ITextInputProps, ITextInput} from './types';

// eslint-disable-next-line react/display-name
const TextInput = forwardRef<ITextInput, ITextInputProps>((props, ref) => (
  <DefaultTextInput
    ref={ref}
    allowFontScaling={getConfig().allowFontScaling}
    {...props}
  />
));
export default TextInput;
