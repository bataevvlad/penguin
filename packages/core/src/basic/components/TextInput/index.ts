import TextInput from './TextInput';

export * from './types';

export {TextInput};
