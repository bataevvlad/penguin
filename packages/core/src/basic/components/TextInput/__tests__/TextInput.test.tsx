import {render} from '@testing-library/react-native';

import TextInput from '../TextInput';

describe('@penguin/core/TextInput', () => {
  it('renders correctly', () => {
    const {toJSON} = render(<TextInput />);
    expect(toJSON()).toMatchSnapshot();
  });
});
