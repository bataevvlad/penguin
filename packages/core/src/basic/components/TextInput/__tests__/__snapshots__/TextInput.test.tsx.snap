// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@penguin/core/TextInput renders correctly 1`] = `
<TextInput
  allowFontScaling={true}
/>
`;
